Shared/Samples/Apply mosaic rule to rasters/ApplyMosaicRuleToRastersView.swift

```diff
@@ -0,0 +1,211 @@
+            centerRule.mosaicMethod = .center
+            centerRule.mosaicOperation = .blend
+            return centerRule
+        case .byAttribute:
+            // A mosaic rule object with byAttribute method and sort on "OBJECTID" field of the service.
+            let byAttributeRule = MosaicRule()
+            byAttributeRule.mosaicMethod = .attribute
+            byAttributeRule.sortField = "OBJECTID"
+            return byAttributeRule
+        case .lockRaster:
+            // A mosaic rule object with lockRaster method and locks 3 image rasters.
+            let lockRasterRule = MosaicRule()
+            lockRasterRule.mosaicMethod = .lockRaster
+            lockRasterRule.addLockRasterIDs([1, 7, 12])
+            return lockRasterRule
+        }
+    }
+    
+    static func == (lhs: RuleSelection, rhs: RuleSelection) -> Bool {
+        return lhs.label == rhs.label
+    }
+}
+
+private extension ApplyMosaicRuleToRastersView {
+    @MainActor
+    class Model: ObservableObject {
+        /// Holds the reference to the currently selected rule.
+        @Published var ruleSelection: RuleSelection = .objectID
+        
+        /// A map with viewpoint set to Amberg, Germany.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISTopographic)
+            map.initialViewpoint = Viewpoint(
+                center: Point(
+                    x: 1320141.0228999995,
+                    y: 6350455.22399999
+                ),
+                scale: 25000.0
+            )
+            return map
+        }()
+        
+        /// A service that fetches the raster using image source url.
+        let imageServiceRaster = {
+            let serviceRaster = ImageServiceRaster(
+                url: .imageServiceURL
+            )
+            serviceRaster.mosaicRule = MosaicRule()
+            return serviceRaster
+        }()
+        
+        init() {
+            let rasterLayer = RasterLayer(raster: imageServiceRaster)
+            map.addOperationalLayer(rasterLayer)
+        }
+        
+        /// A helper function to update the imageService mosiac rule on selection.
+        /// - Parameter rule: The rule selected to update the raster.
+        func updateMosiacRule(with rule: MosaicRule) {
+            imageServiceRaster.mosaicRule = rule
+        }
+        
+        /// A helper function returns center of raster so that map recenters on rule change.
+        /// - Returns: A point which is the center of the raster.
+        func getCenter() -> Point {
+            return imageServiceRaster.serviceInfo?.fullExtent?.center ?? Point(x: 0, y: 0)
+        }
+    }
+}
+
+private extension URL {
+    /// This sample uses a raster image service that shows aerial images of Amberg, Germany.
+    static let imageServiceURL = URL(string: "https://sampleserver7.arcgisonline.com/server/rest/services/amberg_germany/ImageServer")!
+}
+
+#Preview {
+    ApplyMosaicRuleToRastersView()
+}
```

Comment:
```suggestion
#Preview {
    NavigationStack {
        ApplyMosaicRuleToRastersView()
    }
}
```

Don't forgot to add a `NavigationStack` if you use a toolbar, so that it shows up in the preview.